PKHeX/Util/ReflectUtil.cs

```diff
@@ -38,5 +38,26 @@ internal static bool HasProperty(this Type type, string name)
         {
             return type.GetProperties(BindingFlags.Public | BindingFlags.Instance).Any(p => p.Name == name);
         }
+
+        internal static object ConvertValue(object value, Type type)
+        {
+            if (type == typeof(DateTime?)) // Used for PKM.MetDate and other similar properties
+            {
+                DateTime dateValue;
+                if (DateTime.TryParse(value.ToString(), out dateValue))
+                {
+                    return new DateTime?(dateValue);
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
```

Comment:
If there's ever any other types that Convert.ChangeType can't handle, here's where an "else if" for that type should be added.